include/rocksdb/slice.h

```diff
@@ -116,6 +118,52 @@ class Slice {
   // Intentionally copyable
 };
 
+class PinnableSlice : public Slice, public Cleanable {
+ public:
+  PinnableSlice() {}
+
+  inline void PinSlice(const Slice& s, CleanupFunction f, void* arg1,
+                       void* arg2) {
+    data_ = s.data();
+    size_ = s.size();
+    RegisterCleanup(f, arg1, arg2);
+  }
+
+  inline void PinSlice(const Slice& s, Cleanable* cleanable) {
+    data_ = s.data();
+    size_ = s.size();
+    cleanable->DelegateCleanupsTo(this);
+  }
+
+  inline void PinHeap(std::string* s) {
+    data_ = s->data();
+    size_ = s->size();
+    RegisterCleanup(ReleaseStringHeap, s, nullptr);
+  }
+
+  inline void PinSelf(const Slice& slice) {
+    self_space.assign(slice.data(), slice.size());
+    data_ = self_space.data();
+    size_ = self_space.size();
+  }
+
+  inline void PinSelf() {
```

Comment:
good point. IsPinned meant to say if there is any cleanup attached to this. but after PinSelf it has become confusing. i am thinking of removing it entirely. would that work?